internal/agent/list.go

```diff
@@ -12,41 +12,86 @@ import (
 var agentListStyle = lipgloss.NewStyle().Margin(1, 2)
 
 type AgentListModel struct {
-	agentList   list.Model
-	agentLoader tea.Cmd
+	agentList        list.Model
+	agentCurrentPage int
+	agentPerPage     int
+	agentLastPage    int
+	agentsLoading    bool
+	agentLoader      func(int, int) tea.Cmd
 }
 
-func NewAgentList(loader tea.Cmd) AgentListModel {
+func NewAgentList(loader func(int, int) tea.Cmd, page, perpage int) AgentListModel {
```

Comment:
I think its a sane default to just start at the first page. So no need for that argument

Reply:
Removed the perpage argument - I think it was superfluous given that its set upon the `agent list` command run. The `page` is still needed though to pass to the `AgentListOptions` in `loader` 👍